styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -387,6 +388,7 @@ static PersistentWorkflowInstanceState readPersistentWorkflowInstanceState(Entit
           .executionId(readOpt(entity, PROPERTY_STATE_EXECUTION_ID))
           .executionDescription(readOptJson(entity, PROPERTY_STATE_EXECUTION_DESCRIPTION,
               ExecutionDescription.class))
+          .resourceIds(readOpt(entity, PROPERTY_STATE_RESOURCE_IDS))
```

Comment:
This is for reading from Datastore but writing to Datastore is missing (i.e. `activeStateToEntity`)
EDIT: `testFullPersistentStatePersistence` would have caught the issue, also update `FULL_PERSISTENT_STATE` in tests.